### DISCUSSION TITLE
How can I render Drawer component in current dom?

### DISCUSSION THREAD

**Discussion by zhoudalang** [chakra-ui/chakra-ui#7084] [Q&A]
Like antd's Drawer has a getContainer prop?

**Comment by veloware** [comment]
Drawers are mostly Modals under the hood, which use portals, so you should be able to change the portal container ref using the Drawer's portalProps prop. I haven't used it before, but it looks like you should be able to do something like -

const drawerContainerRef = React.useRef()
...
<Box ref={drawerContainerRef}> ** The container that you want to target ** </Box>
...
<Drawer portalProps={{ containerRef: modalContainerRef}}>...</Drawer>

**Reply by zhoudalang** [reply]
Thank you! It woks on Elements tree but it didn't work in fronted display. Drawer slides out from window instead of container component.

**Reply by dennemark** [reply]
I managed to get it to work for a modal. Drawers have disadvantage that they would still slide in from viewport side. But approach might still be working for drawer, too.

https://play.chakra-ui.com/playground?code=N4Igxg9gJgpiBcICWBbADhATgFwATAB0A7XXAIQgA8AaY0sgV22wiNpNwElIjHnX2pAApZsAQwA2g3AFlok6XKiSKUAJ6L5E1WqGYIaAM6blEgMKtsMIthOSzEiIZh8WbOrK0AxCBCuY7CQAJGDFYAI8lSQB5ADcYTAkxDUitPQNjDxEcSXSjaQZnABEkQzBHQwZMGGIAX1wAM30UXAByAAEwAAsxAGtMMQBaBiQAemqxMGxW4lQMHHxcACVQqYA5aBhqXEKYAFEGhpgp7d2Vhtx6pogW1ompmaI50UWwCats8QlLxua2++wgygN0ezwWwGWSAAgph9AB3AAyMAa2AAygikEQYD9rrcAYMkDxDOMkI9iNg1GhsZwiBJMTAohI8oZcABeTymZm4ABk+A8XVC4QA-PBlqtsBtYB5ukgJFBqkRRStJhLNh4IPFEskRbgAEa+CShIgeNA6xk6ZkAbVaaFaAF0PIYkAAvGBmtL6IzWp2u+3SyxiemYc46z65T2Ga08cRB85+oi1YjEUYAKhTdBTXFp9I5kgzuDhsu+Bms2zhApIpWiVJI7OwmAYNRImcMXQgDDleuxuygBaQ2C6uG6fQGLN2JTKFSq2MxuDQYgVeEg6FY1mw+ZYQ8s+mLNdwhnEVnz7XnAxap7EKAADMfqtgqkRDBnRsQYJR5ngGgwiFMkKws3SWKMgAFIQHDRoGWLBsi0gCmECTSDKcoKtIVY1tIrAOE4WzqpqSQpBwaDSD6OEcAAdBRaARnUoo0oBDIehkACUfLgawB64NUFzsmcyIADxBAAKjICIlLEeyGiga4AHzAUQHYSExSYcHeD64MBHikHx0maaQuB8Yyul6fgFFkVRGSJhwxmkGUO7aDAPSxH+mCsgQyCPkgUogEZemlGYa4JDAUA+TZLowKywAkZZ1l6R+4YZBFEGxsiuBCq8AbJQ0opJVB5w-KK36wA09JQNFMW4PWYhoD4YCFBFDSSM4ZUxWh1gRa1CYhbgmEVOFwA9dhzXWSgfh-kQegwM42CuSAhh0rANIUPwKBuT5OlWdZ-V4ckuAAISsuyDUSM4PK8gZWhxAk+Fzk4-ZjRFrRiLqhgQBITAwK09SFlAA4zQAjFeV4AKRubgApIAA5l001uQDwOg6M0lDcZ52mBYNhrl1pAoGoD1iEwECfVjW42JBCTMhFYHldZGBOtgY2io9z2ve9rTSNTpDfQOjNw0DbPE6Q4NQ9gPOA3z7Mcz1hK9KKwEsay0msRz1kDc4csC5cEsxbUyN6et1PAEh8rWLrWmjIy6NWDY+vWXx5taDbZs20pnXGli74vESeBQmgaBsup8uK1TJMccAHXbKw1alt1RBYU1-vjqU5RONO6tsY+eBcQnzjnGnpCjKMN102NnEwEk9PxDH5BULgpS1yuOQ2GRHgF7gjUQAB9KMrgWJBSyYilxcm5xd8OUJC3hcwP2AqYAWAx+5i9G4CNpi1zYHcDyP3Wz-Ohgsv2bcsmP0ENBPbdQLEYikxD2IQBcW+lNsA5iHgajtkOV87Cdi-0ivkht8dDus4B5Gy3CuLENgz5X17AfQsEhvi6mxGAF+3Qgp6jULXbALJTxrhujkb4YgWStggHCEgM8YDNxUjAe8mASDaR8nxCglBB4RS4vUWmd1WAzWqOXJA8RQa6ghhFNydJha6jejUEAX0IoADYrz1C6LI+RjtrJiQYaMJhKitKuH-FLMAvQIqRxrLUaSUciB2x0UQLR+kwyjwyrlZErDkQmK6nxOiXctC10MGY9q3jjExx8f1IgZj6iqz6mElxG0Yqo3sL1SxuBEbExicEQUCRrExRCPBTAST7amEyeEdJelkk6EKcZHQOTzTQDUIUu27igIOyMnbWxKi7brUsspKRQA
